website/oauth/models/__init__.py

```diff
@@ -443,3 +443,59 @@ def absolute_api_v2_url(self):
     # used by django and DRF
     def get_absolute_url(self):
         return self.absolute_api_v2_url
+
+class ApiOAuth2PersonalToken(StoredObject):
+    """Information for user-created personal access tokens
+
+    This collection is also used by CAS to create the master list of available tokens.
+    Any changes made to field names in this model must be echoed in the CAS implementation.
+    """
+    _id = fields.StringField(primary=True,
+                             default=lambda: str(ObjectId()))
+
+    token_id = fields.StringField(default=functools.partial(random_string, length=70),
+                               unique=True)
+
+    owner = fields.ForeignField('User',
+                                backref='created',
+                                index=True,
+                                required=True)
+
+    name = fields.StringField(required=True, index=True)
+
+    scopes = fields.StringField(required=True)
+
+    is_active = fields.BooleanField(default=True, index=True)
+
+    def deactivate(self, save=False):
+        """
+        Deactivate an ApiOAuth2PersonalToken
+
+        Does not delete the database record, but hides this instance from API
+        """
+        client = cas.get_client()
+        # Will raise a CasHttpError if deletion fails, which will also stop setting of active=False.
+        resp = client.revoke_personal_token(self.token_id)  # noqa
+
+        self.is_active = False
+
+        if save:
+            self.save()
+        return True
+
+    @property
+    def url(self):
+        return '/settings/tokens/{}/'.format(self._id)
```

Comment:
If you can't do that unit test, then do a test that checks web_url_for of that method and, if it gets moved from what you've hardcoded, reports where the change needs to be made.